parsl/executors/high_throughput/mpi_prefix_composer.py

```diff
@@ -18,13 +18,19 @@ def __str__(self):
         return f"Invalid resource specification options supplied: {self.invalid_keys}"
 
 
-def validate_resource_spec(resource_spec: Dict[str, str]):
+def validate_resource_spec(resource_spec: Dict[str, str], is_mpi_enabled: bool = False):
     """Basic validation of keys in the resource_spec
 
     Raises: InvalidResourceSpecification if the resource_spec
         is invalid (e.g, contains invalid keys)
     """
     user_keys = set(resource_spec.keys())
+
+    # empty resource_spec when mpi_mode is set causes parsl to hang
+    # ref issue #3427
+    if is_mpi_enabled and len(user_keys) == 0:
+        raise InvalidResourceSpecification({'mpi_mode requires parsl_resource_specification to be configured'})
```

Comment:
InvalidResourceSpecification takes one parameter, which is a set of invalid resource keys. The parameter on this line is not a set of invalid resource keys!